app/models/principal.rb

```diff
@@ -71,8 +71,32 @@ def full_name
     "#{first_name} #{last_name}"
   end
 
+  def next_onboarding_action
+    registered_firms = main_firm_with_trading_names.registered
+
+    if registered_firms.empty?
+      :complete_a_firm
+    elsif needs_advisers?(registered_firms)
+      :complete_an_adviser
+    else
+      :onboarded
+    end
+  end
+
   private
 
+  def needs_advisers?(firm_list)
+     no_remote_firms?(firm_list) && no_advisers_for?(firm_list)
```

Comment:
Knowing the logic for this it still took me a good while to get my head around. The negative `no_` aspect, flipping what could be a positive boolean expression, trips me up in my thinking. Perhaps I'm alone in this. Could it be expressed more positively?